### PR TITLE
ACD-4122: Added test case to check availability of CIM and Eventgen Add-on

### PR DESCRIPTION
**Pull request by rinkaldomadiya-crest**
- Added test case to check availability of CIM and Eventgen Add-on.
- By using the pytest parameter `-m "cim"`, the user can check the CIM availability.
- By using the pytest parameter `-m "field"` or `-m "not cim"`, user can check the Eventgen availability. 
- This type of error will be displayed to the user in case of the addon is not installed in the Splunk.
eventgen_cim

### CODE REVIEW

File: pytest_splunk_addon/standard_lib/addon_basic.py [hunk 1]
```diff
@@ -7,10 +7,49 @@
...
+        self.logger.info(f"Executing the search query: {search}")
+        record_property("search", search)
+
+        result = splunk_search_util.checkQueryCountIsGreaterThanZero(
+            search, interval=INTERVAL, retries=RETRIES
+        )
+
+        assert (
+            result
+        ), f"\nMessage:\nApp {app_name} is not installed/enabled in this Splunk instance. \nThe plugin requires the {app_name} to be installed/enabled in the Splunk instance. Please install the app and execute the tests again"
```
Comment (jayjoshi-crest) [inline pytest_splunk_addon/standard_lib/addon_basic.py:55]: Please format the string in multi-line.

Comment (rinkaldomadiya-crest) [reply pytest_splunk_addon/standard_lib/addon_basic.py]: Done

**Comment by rinkaldomadiya-crest** [comment]
Hello @rfaircloth-splunk 

I have pulled your branch code (include CIM in image) and run the CIM related test case but it's not working correctly. It's might be not copied/installed the Splunk_SA_CIM folder properly on docker Splunk.

In the data models page, it's not displayed any of the CIM data models. 
datamodel

File: pytest_splunk_addon/standard_lib/cim_tests/test_templates.py [hunk 2]
```diff
@@ -18,4 +22,31 @@ class CIMTestTemplates(object):
     """
     logger = logging.getLogger("pytest-splunk-addon-cim-tests")
 
-    # TODO: Add Test scenarios 
+    @pytest.mark.parametrize(
+        "app_name", [pytest.param("Splunk_SA_CIM", marks=[pytest.mark.searchtime_cim])]
```
Comment (jayjoshi-crest) [inline pytest_splunk_addon/standard_lib/cim_tests/test_templates.py:26]: Update the marker to "splunk_app_cim".

Comment (jayjoshi-crest) [reply pytest_splunk_addon/standard_lib/cim_tests/test_templates.py:26]: As discussed, we need to update the marker to "splunk_searchtime_cim"

Comment (rinkaldomadiya-crest) [reply pytest_splunk_addon/standard_lib/cim_tests/test_templates.py]: Updated marker name

**Pull request merged**